lib/iris/coords.py

```diff
@@ -210,6 +210,9 @@ def __add__(self, mod):
             bound = tuple([val + mod for val in bound])
         return Cell(point, bound)
 
+    def __hash__(self):
+        return hash(tuple(self))
```

Comment:
I'm not sure it's more efficient, but I think it might be clearer that way. Just running `%timeit hash(b)` with the class in your example above shows `super` to be maybe 50 ns slower, _but_ the slowest time is anywhere from 8 to 20-25 times slower for either method. So there's some sort of caching, but for that first hit, it seems to be a toss-up.